l3kernel/l3basics.dtx

```diff
@@ -3412,11 +3417,50 @@
     \token_if_macro:NTF #1
       {
         \exp_after:wN \@@_prefix_arg_replacement:wN
-          \token_to_meaning:N #1 \s_@@_stop \use_iii:nnn
+          \@@_meaning:N #1 \s_@@_stop \use_iii:nnn
       }
       { \scan_stop: }
   }
 %    \end{macrocode}
+%   LuaMeta\TeX{}'s version of \tn{meaning} doesn't give the same output
+%   as the classical one. So we abstract out the idea at a low level, as
+%   we need the details to be correct here.
+%    \begin{macrocode}
+\cs_if_exist:NF \@@_meaning:N
+  { \cs_new_eq:NN \@@_meaning:N \cs_meaning:N }
+%</code>
+%    \end{macrocode}
+%   This is a modified version of code suggested by Max Chernoff
+%   (\url{https://chat.stackexchange.com/transcript/message/67947633#67947633}).
+%   We aim to provide the classical \TeX{}82 appearance but covering \eTeX{}:
+%   due to changes in the engine, the only useful prefix here is |\protected|
+%   at present. (This may need to be revisited once the engine is more stable.)
+%   This does not cover tokens other than control sequences, so we only use
+%   it for the internal needed here.
+%    \begin{macrocode}
+%<*lua>
+if status.luatex_engine == "luametatex" then
+  luacmd("@@_meaning:N", function()
+    local csname = scan_csname(true)
+    local parameters = get_macro(csname, false, true)
+    local body = get_macro(csname)
```

Comment:
This breaks for active characters.

Reply:
Right, we should be able to use `token.scannext()` (aka `token.get_next()`) instead though.